File: Python/Leetcode/HashMap/isomorphicStrings.py
class Solution(object):
    def isIsmorphic(self, s, t):
        mapST, mapTS = {}, {}
        
        for c1, c2 in zip(s, t):
            
            if ((c1 in mapST and mapST[c1] != c2) or 
                (c2 in mapTS and mapTS[c2] != c1)):
                return False
            
            mapST[c1] = c2
            mapTS[c2] = c1
            
        return True
    
if __name__ == '__main__':
    solution_instance = Solution()
    solution_instance.isIsmorphic("foo", "bar")

File: C++/Leetcode/Hashmap/isomorphicStrings.cpp
#include <iostream>
#include <string>
#include <map>

class Solution {
    public:
        bool isIsomorphic(std::string s, std::string t){ // Not complete need to learn how to zip
            if (s.size() != t.size()){
                return false;
            }
            
            std::map<char, int> sMap;
            std::map<char, int> tMap;
            for (int i=0; i<s.size(); ++i){
                sMap[s[i]]++;
                tMap[t[i]]++;
            }

            for ()
        }

    private:
        void printMap(std::map<char, int> map){
            for (const auto& pair : map){
                std::cout << pair.first << " " << pair.second << std::endl;
            }
        }
};

int main(){
    Solution solution_instance;
    solution_instance.isIsomorphic("egg", "add");
}